reggie/ingestion/preprocessor/west_virginia_preprocessor.py

```diff
@@ -1,87 +1,361 @@
+        and history files.  Should set self.main_file with combined dataframe.
+        """
+        # Shortcut to config
+        config = self.config.data
+
+        # Download if not file not provided
         if self.raw_s3_file is not None:
             self.main_file = self.s3_download()
 
-        new_files = [
-            n for n in self.unpack_files(self.main_file, compression="unzip")
+        # Get the files from the archive
+        new_files = [n for n in self.unpack_files(self.main_file, compression="unzip")]
+
+        # Should only be one voter file
+        voter_file_regex = re.compile(".*statewide.*vr")
+        voter_files = [
+            n for n in new_files if voter_file_regex.match(n["name"].lower())
         ]
+        if len(voter_files) > 1:
+            raise UnexpectedNumberOfFilesError(
+                f"{config['state']} has too many voter files."
+            )
+        if len(voter_files) < 1:
+            raise UnexpectedNumberOfFilesError(
+                f"Unable to find voter file in {config['state']} import."
+            )
+        voter_file = voter_files[0]
 
-        # only one voter file, no voter history
-        voter_file = [n for n in new_files if "wv" in n["name"].lower()][0]
+        # Read voter file into pandas dataframe
+        df_voters = pd.read_csv(
+            voter_file["obj"],
+            sep=config["delimiter"],
+            encoding="latin-1",
+            dtype=str,
+            header=0 if config["has_headers"] else None,
+        )
 
-        df_voter = pd.read_csv(
-            voter_file["obj"], sep="|", encoding="latin-1", dtype=str, header=0
+        # Add county id column
+        df_voters.insert(
+            1,
+            "County_ID",
+            df_voters.County_Name.map(slugify),
         )
 
-        # --- handling voter file --- #
+        # Make sure counties are valid as defined in config
+        # counties = df_voters.County_ID.unique()
+        self.check_column_has_valid_values(
+            df_voters,
+            "County_ID",
+            self.config_lookup_to_valid_list(config["locales_counties"]),
+        )
 
-        gender_conversion_dict = {}
-        for c, v in self.config["gender_codes"].items():
-            for i in v:
-                if i is None:
-                    gender_conversion_dict[" "] = c
-                gender_conversion_dict[i] = c
+        # Gender clean
+        gender_dict = self.config_lookup_to_dict(self.config["gender_codes"])
+        df_voters.loc[:, "SEX"] = df_voters.loc[:, "SEX"].map(gender_dict)
 
-        df_voter.loc[:, "SEX"] = df_voter.loc[:, "SEX"].map(
-            gender_conversion_dict
+        # Gender check
+        self.check_column_has_valid_values(
+            df_voters, "SEX", self.config_lookup_to_valid_list(config["gender_codes"])
         )
 
-        df_voter = self.config.coerce_dates(df_voter)
-        df_voter = self.config.coerce_strings(
-            df_voter, exclude=[self.config["voter_id"]]
```

Comment:
you should probably still call coerce_strings - it'll just remove whitespace and make all the data lowercase (except for party & status fields. You might also want to also exclude the voter ID from that normalization like Delaney did here)